### DISCUSSION TITLE
Immich as gallery for server-stored photos (no upload, no backup)

### DISCUSSION THREAD

**Discussion by andyxpert** [immich-app/immich#4843] [Q&A] [answered]
LE: TL;DR: I had some workarounds to handle image upload and library refresh which now are history because of the new External Library feature ( https://github.com/immich-app/immich/pull/3124 ).
The Android local images appearing in timeline was temporarily "fixed" by only allowing an empty Album to be watched.

Now for the original message (aka the Long Part):
Hello,

First of all, thank you for creating Immich, after trying more than a dozen solutions (from very lightweight file browsers to complicated multi-tiered solutions such as nextcloud), I most probably will continue to use Immich for a long time.

Now, my use-case is fairly simple and I'm sure many have a similar scenario - I have my photo backup on a linux server already structured and organized folder-based (for future-proof and ease of use), I use Immich only as a gallery for this big folder structure to be able to have a Google Photos-like experience but still have some power-user functionalities.
Hands down Immich does this amazingly.

My pain is with the other features which seem to be tightly-bound and not that simple to "deactivate"
For example:
- Why does Immich insist so much that I upload (automatically via Sync and Backup) my local photos ?
- Why is it now showing also my local photos in the timeline alongside my server-stored ones ? (LE: I want to disable upload / backup altogether, not force only an empty album for backup so that i hide device photos - as a workaround)
- Why doesn't it allow me to easily refresh only a folder from the external source ?

Overall - could such features be toggled in a settings screen ? I think feature-toggling would be an amazing feature :) also helping with resource mgmt (currently I kill the ML service to free up resources when it's not needed, but i think it would be waaaay easier to have such toggles built-in a simple list of toggles - similar to the Jobs screen that allows me to run specific jobs when I want/need to)

Most of us are already power users mainly working in IT and have to do workarounds to make Immich do what we want.
My setup is:
- folder based structure (date-based) where I upload photos via some scripts or other 3rd party services, pre-organize them, transcode videos, etc
- folder structure added to Immich as external source (ReadOnly, for peace of mind)
- everytime i add new photos to the drive I run immich upload command (docker) to pick-up the new photos
Now, for me it would be great if I could easily refresh the photos folder, or toggle-off the possibility to upload from immich UI photos, or to only see the photos from the server, not the ones on the device (Android problem , iOS seems to be ok).

Anyway, I want your view on my (and most of us, I guess), am I doing something wrong or too complicated, should I do something different, how could I improve my setup to rely less on workarounds and use more or better Immich features ?

But yeah, great solution, just try not to make it more non-technical ppl friendly and lose the technical touch...
The Arr apps seem to have nailed it, being relatively intuitive but also allowing a lot of complex configuration for who wants to tinker.

Cheers, keep on improving Immich, we thank you!

**Comment by danieldietzler** [comment, marked as answer]
Instead of uploading new photos every now and then you could also look into the [libraries feature](https://immich.app/docs/features/libraries), specifically an external library. That is, if I've understood (part of) your use case correctly.

**Reply by andyxpert** [reply]
https://github.com/immich-app/immich/pull/3124 I last configured it 'bout a year ago, greatly appreciate this feature
Gotta do sone upgrades it seems

Thanks for pointing this out, I only used the basic external library concept, didn't know that there is some level of configuration added to it recently.

**Reply by andyxpert** [reply]
Or at least a split by import paths or a filter by import paths for the library or timeline view; or native "albums" based on import paths

Today I'm rebuilding my library, so not sure whether this is already implemented, though I fear it isn't yet.

**Reply by klaus1k** [reply]
Yes, external libraries are amazing!
What still does not seem possible is a "folder based" view, so that you can browse albums in external library just like the folders are structured. This would be really neat to have. #2073 asks for nested albums, which should solve this issue.
